### PR TITLE
Model type in convert_tf_checkpoint_to_pytorch and 'squad' mapping

### PR DESCRIPTION
**Pull request by mhardalov**
Issue #438 still exists if you choose to use something else rather then BertForTokenClassification. Furthermore, you still need to edit the code before running the convertor. Lastly,  BertForTokenClassification is not the same as BertForQuestionAnswering, since the latter omits the dropout before the output layer.

Maybe it's better to add more options like 'classification' which uses BertForTokenClassification.

Tested the changes on fine-tuned BERT model on SQuAD 1.1 with Google's original Tensorflow script run_squad.py initialized with multi_cased_L-12_H-768_A-12.

### CODE REVIEW

File: pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py [hunk 1]
```diff
@@ -25,13 +25,20 @@
...
+
+def convert_tf_checkpoint_to_pytorch(tf_checkpoint_path, bert_config_file, pytorch_dump_path, bert_model):
     # Initialise PyTorch model
     config = BertConfig.from_json_file(bert_config_file)
     print("Building PyTorch model from configuration: {}".format(str(config)))
-    model = BertForPreTraining(config)
+
+    # Dynamically choose the model type based on input type
+    if bert_model == "qa":
+        model = BertForQuestionAnswering(config)
```
Comment (thomwolf) [inline pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py]: Maybe we could use a more general way to give the class name as a string, like this:
```python
import importlib
ModelClass = getattr(importlib.import_module("pytorch_pretrained_bert.modeling"),
                     bert_model)
model = ModelClass(config)
```
See this: https://stackoverflow.com/questions/4821104/python-dynamic-instantiation-from-string-name-of-a-class-in-dynamically-imported

File: pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py [hunk 2]
```diff
@@ -60,7 +67,15 @@ def convert_tf_checkpoint_to_pytorch(tf_checkpoint_path, bert_config_file, pytor
...
+                        type = str,
+                        required = False,
+                        help = "Type of the model which is going to be converted. \n"
+                               "This specifies the final layer.")
     args = parser.parse_args()
     convert_tf_checkpoint_to_pytorch(args.tf_checkpoint_path,
                                      args.bert_config_file,
-                                     args.pytorch_dump_path)
+                                     args.pytorch_dump_path,
+                                     args.bert_model)
```
Comment (thomwolf) [inline pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py]: Maybe it would be better to add a dictionary of variable conversions here that would be, in this case `{'squad': 'qa_output'}.

This way we can accomodate other type of models like in #580.

This PR should let us also do the workflow of #580 without problems.

**Review by thomwolf — CHANGES_REQUESTED** [review]

File: pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py (hunk 1 above)
Comment (mhardalov) [reply pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py]: My concern with that approach, which indeed is more generic, was the case of having different constructors, e.g. "BertForTokenClassification(config, 2)" (which is the only model, atm). I guess we can do dynamic argument passing as well.

Something like this: [How can you set class attributes from variable arguments (kwargs) in python](https://stackoverflow.com/questions/8187082/how-can-you-set-class-attributes-from-variable-arguments-kwargs-in-python)

Maybe it's better to move the bert_model (instance class name) and constructor arguments into a new config file (e.g. converter_config.json)?

`{"bert_model": "BertForTokenClassification",
"num_labels": 2,
"bert_config_file": "/some/path/here"}`

Several questions:
1) I'm not sure if it's not an overkill, since just one class at the moment has an extra argument in the constructor?
2) Is it a good idea to move bert_config_file into the "config" file?
3) Should we move then all other args to the config?
4) Should we allow "default" execution without converter_config, that instantiates "BertForPreTraining"?

File: pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py (hunk 2 above)
Comment (mhardalov) [reply pytorch_pretrained_bert/convert_tf_checkpoint_to_pytorch.py]: Makes sense, we can add this to the config mentioned in the [discussion above](https://github.com/huggingface/pytorch-pretrained-BERT/pull/521#discussion_r282359014).

**Comment by mhardalov** [comment]
@thomwolf Is this PR still useful? Can it be somehow improved and later merged, or it should be closed?
